lib/spack/spack/cmd/create.py

```diff
@@ -654,8 +679,11 @@ def create(parser, args):
     build_system = get_build_system(args, guesser)
 
     # Create the package template object
+    constr_args = [name, versions]
     package_class = templates[build_system]
-    package = package_class(name, url, versions)
+    if package_class != DependenciesPackageTemplate:
+        constr_args.insert(1, url)
```

Comment:
I generally prefer not to use insertions to modify lists with a prescribed order. This avoids future issues if say the argument API for `Package` changed later. If these were `kwargs` (i.e. a dictionary) it wouldn't be an issue.

Reply:
Fair enough.